jpsonic-main/src/main/java/com/tesshu/jpsonic/service/scanner/ScannerProcedureService.java

```diff
@@ -219,9 +219,7 @@ void beforeScan(@NonNull Instant scanDate) {
         mediaFileCache.setEnabled(false);
         mediaFileCache.removeAll();
 
-        if (!settingsService.isUseCleanUp() && mediaFileDao.existsNonPresent()) {
-            // If useCleanUp=false(default), Clean-up can be managed by Scan processes.
-            // Removing present=false in advance can reduce the number of subsequent queries issued.
+        if (mediaFileDao.existsNonPresent()) {
```

Comment:
Yes, it's unnecessary, but it's only a part of the whole unnecessary thing. It would be better to have it deleted in a planned manner.

Subsonic/Airsonic uses logical deletion, but the drawbacks far outweigh the benefits and it was the cause of many many many many problems. Therefore, Jpsonic has been redesigned to not use logical deletion. So it could be removed, but it's probably best to defer this to a later version.

 - Assuming that the option to use Clean Up is enabled, it would be better for "Delete logical-deletion" to be performed at least in a version later than the Clean Up version.
 - This code isn't the only legacy of logical deletion. There is other trash too.
   - First, present-flag should be removed from all SQL.
   - There is still some crappy code, like the Ajax PlaylistService#populateAccess. (This is not a logical deletion, but is used for other purposes filter.) It will need to be rewritten correctly.
   - We will need to remove this code as well as any clean up code that assumes logical deletion.

Doing this will eliminate the side effects. Maintain status quo.

Reply:
It would be better if a new issue was issued regarding this matter.

Reply:
#2693